File: msdialog/src/main/java/com/maple/msdialog/AlertEditDialog.java
package com.maple.msdialog;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Typeface;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * 警告框式Edit Dialog [ 标题 + 输入框 + 消息文本 + 左按钮 + 右按钮 ]
 *
 * @author maple
 * @time 17/3/23
 */
public class AlertEditDialog extends BaseDialog {
    private TextView txt_title;
    private TextView txt_msg;
    private EditText et_text;
    private Button leftBtn;
    private Button rightBtn;
    private ImageView img_line;

    private boolean showTitle = false;
    private boolean showMsg = false;
    private boolean showLeftBtn = false;
    private boolean showRightBtn = false;

    public AlertEditDialog(Context context) {
        super(context);
        rootView = LayoutInflater.from(context).inflate(R.layout.view_alert_edit_dialog, null);

        // get custom Dialog layout
        txt_title = rootView.findViewById(R.id.txt_title);
        txt_title.setVisibility(View.GONE);
        txt_msg = rootView.findViewById(R.id.txt_msg);
        txt_msg.setVisibility(View.GONE);
        et_text = rootView.findViewById(R.id.et_text);
        leftBtn = rootView.findViewById(R.id.btn_left);
        leftBtn.setVisibility(View.GONE);
        rightBtn = rootView.findViewById(R.id.btn_right);
        rightBtn.setVisibility(View.GONE);
        img_line = rootView.findViewById(R.id.img_line);
        img_line.setVisibility(View.GONE);

        // set Dialog style
        dialog = new Dialog(context, R.style.AlertDialogStyle);
        dialog.setContentView(rootView);

        setScaleWidth(0.85);
    }

    public AlertEditDialog setCancelable(boolean cancel) {
        dialog.setCancelable(cancel);
        return this;
    }

    @Override
    public AlertEditDialog setScaleWidth(double scWidth) {
        return (AlertEditDialog) super.setScaleWidth(scWidth);
    }

    public AlertEditDialog setTitle(String title) {
        int color = ContextCompat.getColor(mContext, R.color.def_title_color);
        return setTitle(title, color, 18, true);
    }

    public AlertEditDialog setTitle(String title, int color, int size, boolean isBold) {
        showTitle = true;
        txt_title.setText(title);
        if (color != -1) {
            txt_title.setTextColor(color);
        }
        if (size > 0) {
            txt_title.setTextSize(size);
        }
        if (isBold) {
            txt_title.setTypeface(txt_title.getTypeface(), Typeface.BOLD);
        }
        return this;
    }

    public AlertEditDialog setMessage(String msg) {
        int color = ContextCompat.getColor(mContext, R.color.def_title_color);
        return setMessage(msg, color, 16, false);
    }

    public AlertEditDialog setMessage(String msg, int color, int size, boolean isBold) {
        showMsg = true;
        txt_msg.setText(msg);
        if (color != -1) {
            txt_msg.setTextColor(color);
        }
        if (size > 0) {
            txt_msg.setTextSize(size);
        }
        if (isBold) {
            txt_msg.setTypeface(txt_msg.getTypeface(), Typeface.BOLD);
        }
        return this;
    }

    public AlertEditDialog setRightButton(String text, final EditTextCallListener listener) {
        int color = ContextCompat.getColor(mContext, R.color.def_title_color);
        return setRightButton(text, color, 16, false, listener);
    }

    public AlertEditDialog setRightButton(String text, int color, int size, boolean isBold, final EditTextCallListener listener) {
        showRightBtn = true;
        rightBtn.setText(text);
        if (color != -1) {
            rightBtn.setTextColor(color);
        }
        if (size > 0) {
            rightBtn.setTextSize(size);
        }
        if (isBold) {
            rightBtn.setTypeface(rightBtn.getTypeface(), Typeface.BOLD);
        }
        rightBtn.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null) {
                    String inputText = et_text.getText().toString().trim();
                    listener.callBack(inputText);
                }
                dialog.dismiss();
            }
        });
        return this;
    }

    public AlertEditDialog setLeftButton(String text, final OnClickListener listener) {
        int color = ContextCompat.getColor(mContext, R.color.def_title_color);
        return setLeftButton(text, color, 16, false, listener);
    }

    public AlertEditDialog setLeftButton(String text, int color, int size, boolean isBold, final OnClickListener listener) {
        showLeftBtn = true;
        leftBtn.setText(text);
        if (color != -1) {
            leftBtn.setTextColor(color);
        }
        if (size > 0) {
            leftBtn.setTextSize(size);
        }
        if (isBold) {
            leftBtn.setTypeface(leftBtn.getTypeface(), Typeface.BOLD);
        }
        leftBtn.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null) {
                    listener.onClick(v);
                }
                dialog.dismiss();
            }
        });
        return this;
    }

    public interface EditTextCallListener {
        void callBack(String str);
    }

    private void setLayout() {
        if (!showTitle && !showMsg) {
            txt_title.setText(R.string.alert);
            txt_title.setVisibility(View.VISIBLE);
        }

        if (showTitle) {
            txt_title.setVisibility(View.VISIBLE);
        }

        if (showMsg) {
            txt_msg.setVisibility(View.VISIBLE);
        }

        if (!showRightBtn && !showLeftBtn) {
            rightBtn.setText(R.string.ok);
            rightBtn.setVisibility(View.VISIBLE);
            rightBtn.setBackgroundResource(R.drawable.alertdialog_single_selector);
            rightBtn.setOnClickListener(new OnClickListener() {
                @Override
                public void onClick(View v) {
                    dialog.dismiss();
                }
            });
        }

        if (showRightBtn && showLeftBtn) {
            rightBtn.setVisibility(View.VISIBLE);
            rightBtn.setBackgroundResource(R.drawable.alertdialog_right_selector);
            leftBtn.setVisibility(View.VISIBLE);
            leftBtn.setBackgroundResource(R.drawable.alertdialog_left_selector);
            img_line.setVisibility(View.VISIBLE);
        }

        if (showRightBtn && !showLeftBtn) {
            rightBtn.setVisibility(View.VISIBLE);
            rightBtn.setBackgroundResource(R.drawable.alertdialog_single_selector);
        }

        if (!showRightBtn && showLeftBtn) {
            leftBtn.setVisibility(View.VISIBLE);
            leftBtn.setBackgroundResource(R.drawable.alertdialog_single_selector);
        }
    }

    public void show() {
        setLayout();
        dialog.show();
    }

}

File: msdialog/src/main/java/com/maple/msdialog/AlertEditDialog.kt
package com.maple.msdialog

import android.content.Context
import android.graphics.Typeface
import android.os.Build
import android.text.Html
import android.text.Spanned
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import androidx.annotation.ColorInt
import androidx.annotation.StyleRes
import androidx.core.content.ContextCompat
import com.maple.msdialog.databinding.MsDialogAlertEditBinding
import com.maple.msdialog.utils.DensityUtils.dp2px

/**
 * 警告框式Edit Dialog [ 标题 + 消息文本 + 输入框 + 左按钮 + 右按钮 ]
 *
 * @author shaoshuai
 * @time 2017/3/23
 */
class AlertEditDialog(
    private val mContext: Context,
    private val config: Config = Config(mContext),
    @StyleRes themeResId: Int = R.style.AlertDialogStyle
) : AlertBaseDialog(mContext, config, themeResId) {
    private val binding: MsDialogAlertEditBinding = MsDialogAlertEditBinding.inflate(
        LayoutInflater.from(mContext), null, false
    )
    private var showTitle = false
    private var showMsg = false

    // 方便java调用
    constructor(mContext: Context) : this(mContext, Config(mContext), R.style.AlertDialogStyle)
    constructor(mContext: Context, config: Config) : this(mContext, config, R.style.AlertDialogStyle)

    fun getTitleView() = binding.tvTitle
    fun getMessageView() = binding.tvMsg

    fun setDialogTitle(title: CharSequence?): AlertEditDialog {
        return setTitle(title, isBold = false)
    }

    override fun setTitle(title: CharSequence?) {
        super.setTitle(title)
        this.setTitle(title, isBold = false)
    }

    fun setTitle(
        title: CharSequence?,
        @ColorInt color: Int = config.titleColor,// 字体颜色
        spSize: Float = config.titleTextSizeSp,// 字体大小
        isBold: Boolean = false// 是否加粗
    ): AlertEditDialog {
        showTitle = true
        getTitleView().apply {
            text = title ?: config.defNullText
            setTextColor(color)
            textSize = spSize
            setPadding(config.titlePaddingLeft, config.titlePaddingTop, config.titlePaddingRight, config.titlePaddingBottom)
            setTypeface(typeface, if (isBold) Typeface.BOLD else Typeface.NORMAL)
        }
        return this
    }

    fun setHtmlMessage(message: String?): AlertEditDialog {
        return setMessage(convertHtmlText(message))
    }

    /**
     * 将html类文本转换成普通文本
     */
    fun convertHtmlText(htmlText: String?): Spanned {
        val source = htmlText ?: (config.defNullText) as String
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            Html.fromHtml(source, Html.FROM_HTML_MODE_LEGACY)
        } else {
            Html.fromHtml(source)
        }
    }

    fun setMessage(message: CharSequence?) = setMessage(message, isBold = false)

    fun setMessage(
        message: CharSequence?,
        @ColorInt color: Int = config.messageColor,// 字体颜色
        spSize: Float = config.messageTextSizeSp, // 字体大小
        isBold: Boolean = false, // 是否加粗
        gravity: Int = Gravity.CENTER// 偏左，居中，偏右
    ): AlertEditDialog {
        showMsg = true
        getMessageView().apply {
            text = message ?: config.defNullText
            setTextColor(color)
            this.gravity = gravity
            textSize = spSize
            setPadding(config.messagePaddingLeft, config.messagePaddingTop, config.messagePaddingRight, config.messagePaddingBottom)
            setTypeface(typeface, if (isBold) Typeface.BOLD else Typeface.NORMAL)
        }
        return this
    }

    fun setRightButton(
        text: CharSequence?,
        listener: OnEditTextCallListener? = null
    ) = setRightButton(text, listener, config.rightBtnColor, config.rightBtnTextSizeSp, false)

    fun setRightButton(
        text: CharSequence?,
        listener: OnEditTextCallListener? = null,
        @ColorInt color: Int = config.rightBtnColor,
        spSize: Float = config.rightBtnTextSizeSp,
        isBold: Boolean = false
    ): AlertEditDialog {
        showRightBtn = true
        getRightBtnView().apply {
            this.text = text ?: config.defNullText
            setTextColor(color)
            textSize = spSize
            setTypeface(typeface, if (isBold) Typeface.BOLD else Typeface.NORMAL)
            setOnClickListener {
                if (listener != null) {
                    val inputText = binding.etText.text.toString().trim()
                    listener.callBack(inputText)
                }
                dismiss()
            }
        }
        return this
    }

    override fun setLayout() {
        super.setLayout()
        setDialogContext(binding.root)
        // title & msg
        getTitleView().visibility = if (showTitle) View.VISIBLE else View.GONE
        getMessageView().visibility = if (showMsg) View.VISIBLE else View.GONE
        if (!showTitle && !showMsg) {
            getTitleView().text = config.defNullText
            getTitleView().visibility = View.VISIBLE
        }
    }

    /**
     * AlertEditDialog 的配置
     */
    class Config(
        context: Context
    ) : AlertBaseDialog.Config(context) {
        // title
        var titleTextSizeSp: Float = 18f // 字体大小
        @ColorInt var titleColor: Int = ContextCompat.getColor(context, R.color.ms_def_title_color) // 字体颜色
        var titlePaddingLeft: Int = 15f.dp2px(context) // 左边距
        var titlePaddingTop: Int = 15f.dp2px(context) // 上边距
        var titlePaddingRight: Int = 15f.dp2px(context) // 右边距
        var titlePaddingBottom: Int = 0f.dp2px(context) // 下边距

        // message
        var messageTextSizeSp: Float = 16f // 字体大小
        @ColorInt var messageColor: Int = ContextCompat.getColor(context, R.color.ms_def_message_color) // 字体颜色
        var messagePaddingLeft: Int = 15f.dp2px(context) // 左边距
        var messagePaddingTop: Int = 15f.dp2px(context) // 上边距
        var messagePaddingRight: Int = 15f.dp2px(context) // 右边距
        var messagePaddingBottom: Int = 5f.dp2px(context) // 下边距
    }

}